x/gamm/twap/keeper.go

```diff
@@ -1,8 +1,27 @@
 package twap
 
-import sdk "github.com/cosmos/cosmos-sdk/types"
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
 
-type twapkeeper struct {
-	// storeKey     sdk.StoreKey
-	transientKey sdk.TransientStoreKey
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+
+	"github.com/osmosis-labs/osmosis/v10/x/gamm/twap/types"
+)
+
+type Keeper struct {
+	storeKey     sdk.StoreKey
+	transientKey *sdk.TransientStoreKey
+
+	paramSpace paramtypes.Subspace
+
+	ammkeeper types.AmmInterface
+}
+
+func NewKeeper(storeKey sdk.StoreKey, transientKey *sdk.TransientStoreKey, paramSpace paramtypes.Subspace, ammKeeper types.AmmInterface) *Keeper {
+	return &Keeper{storeKey: storeKey, transientKey: transientKey, paramSpace: paramSpace, ammkeeper: ammKeeper}
+}
+
+// TODO: make this read from a parameter, or hardcode it.
+func (k *Keeper) PruneEpochIdentifier(ctx sdk.Context) string {
+	return "daily"
```

Comment:
Should it be "day" and not "daily" according to the epochs parameter?

Reply:
oh woops, yeah it should